model/keyvalue.go

```diff
@@ -272,6 +277,18 @@ func (kvs KeyValues) Equal(other KeyValues) bool {
 	return true
 }
 
+// FindMatch checks for a matching KeyValue in the KeyValues for the given
+// query key/value pair. Returns the match and true when the match is found.
+// Returns empty KeyValue and false when no match is found.
+func (kvs KeyValues) FindMatch(key, value string) (KeyValue, bool) {
```

Comment:
I can't see on the phone but I thought we already had a find-like method for a list of tags. Also, the method seems suspicious given that it takes bake as string.

Reply:
There was a note in the existing code explaining the `FindKey` method only handles one value rather than multiple.